app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/EmojiConverter.java

```diff
@@ -56,17 +61,30 @@
             {"\u2764", "<3"},         // heart
     };
 
+    private static boolean isInitialised = false;
+
     private static String convertSimpleEmojiToAscii(String text) {
         for (String[] emojiMap : simpleEmojiMapping) {
             text = text.replace(emojiMap[0], emojiMap[1]);
         }
         return text;
     }
 
-    public static String convertUnicodeEmojiToAscii(String text) {
+    private static String convertAdvancedEmojiToAscii(String text, Context context) {
```

Comment:
```suggestion
     private static synchronized String convertAdvancedEmojiToAscii(String text, Context context) {
```
to prevent concurrency issues in case this is ever called from multiple threads in the future.